### PR TITLE
:pencil2: add OP_SWAP

### PR DESCRIPTION
**Pull request by akifuj**
Added OP_SWAP

### CODE REVIEW

File: BitcoinCashKit/OP_SWAP.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+public struct OpSwap: OpCodeProtocol {
+    public var value: UInt8 { return 0x7c }
+    public var name: String { return "OP_SWAP" }
+    
+    // (x1 x2 -- x2 x1)
+    public func execute(_ context: ScriptExecutionContext) throws {
+        try prepareExecute(context)
+        guard context.stack.count >= 2 else {
+            throw OpCodeExecutionError.opcodeRequiresItemsOnStack(2)
+        }
```
Comment (usatie) [inline BitcoinCashKit/OP_SWAP.swift]: Please use `context.assertStackHeightGreaterThan(2)` instead!

**Comment by usatie** [comment]
Please edit `OpCode.swift` as well.


```
public static let OP_SWAP: OpCodeProtocol = OpExample()
```

->

```
public static let OP_SWAP: OpCodeProtocol = OpSwap()
```

**Comment by usatie** [comment]
@GitCash send 1000 bits to @akifuj

**Comment by GitCash** [comment]
Hey **akifuj**, user **usatie** tipped you **1000 bits** in Bitcoin Cash ( ~ **$0.821** ).

*[Click here to claim it!](https://gitcash.io/claim/akifuj)*

You can also add the "thumbs down" reaction to **usatie**'s comment above to prevent future tips.

**Pull request merged**